controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -258,6 +260,28 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.upgradeControlPlane(ctx, cluster, kcp, ownedMachines, requireUpgrade)
 	}
 
+	ds := &v1.DaemonSet{}
+	err = r.Client.Get(ctx, client.ObjectKey{Namespace: "kube-system", Name: "kube-proxy"}, ds)
+	if err != nil && apierrors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	} else if err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
hm, i think i saw this in another PR today, maybe there is a pattern in CAPI.
this could be:
```
if err := r.Client.Get(ctx, client.ObjectKey{Namespace: "kube-system", Name: "kube-proxy"}, ds); err != nil {
	if apierrors.IsNotFound(err) {
		return ctrl.Result{}, nil
	}
	return ctrl.Result{}, err
}
```
though the number of LOC is the same.

Reply:
fwiw the pattern you have here reads better imo